File: garden_advice.js
// Function to get user input with validation
function getValidInput(prompt, validOptions) {
    let userInput;
    do {
        userInput = prompt(prompt).toLowerCase().trim();
        if (!validOptions.includes(userInput)) {
            alert(`Invalid option. Please choose from: ${validOptions.join(', ')}`);
        }
    } while (!validOptions.includes(userInput));
    return userInput;
}

// Function to generate gardening advice
function generateAdvice(season, plantType) {
    const adviceDict = {
        "summer": {
            "flower": "Water your plants regularly, provide shade, and use fertiliser to encourage blooms.",
            "vegetable": "Water your plants regularly and keep an eye out for pests!"
        },
        "winter": {
            "flower": "Protect your plants from frost with covers and use fertiliser sparingly.",
            "vegetable": "Protect your plants from frost and monitor for pests indoors."
        }
    };
    
    return adviceDict[season]?.[plantType] || "No specific advice for this combination.";
}

// Main function to run the garden advice app
function runGardenAdvice() {
    console.log("🌱 Welcome to the Gardening Advice App! 🌱");
    
    const seasons = ["summer", "winter"];
    const plantTypes = ["flower", "vegetable"];
    
    // Get user input
    const season = getValidInput("Enter the current season (summer/winter):", seasons);
    const plantType = getValidInput("Enter the type of plant (flower/vegetable):", plantTypes);
    
    // Generate and display advice
    const advice = generateAdvice(season, plantType);
    console.log("\nYour Gardening Advice:");
    console.log(advice);
    
    // Display advice in the HTML interface
    displayAdvice(advice);
}

// Function to display advice in the HTML interface
function displayAdvice(advice) {
    const adviceDisplay = document.getElementById('advice-display');
    const adviceText = document.getElementById('advice-text');
    
    adviceText.textContent = advice;
    adviceDisplay.style.display = 'block';
    
    // Scroll to the advice
    adviceDisplay.scrollIntoView({ behavior: 'smooth' });
}

// Initialize the app when the page loads
document.addEventListener('DOMContentLoaded', function() {
    console.log("Garden Advice App loaded successfully!");
});

// TODO: Examples of possible features to add:
// - Add detailed comments explaining each block of code.
// - Refactor the code into functions for better readability and modularity.
// - Store advice in an object for multiple plants and seasons.
// - Suggest plants that thrive in the given season.

File: garden_advice.py
"""
A simple command-line application that provides gardening advice
based on the season and type of plant.
"""

# garden_advice.py


def get_user_input(prompt: str, valid_options: list[str]) -> str:
    """
    Prompt the user for input and validate it against allowed options.

    Args:
        prompt (str): The message displayed to the user.
        valid_options (list[str]): List of valid input options.

    Returns:
        str: Validated user input.
    """
    while True:
        user_input = input(prompt).strip().lower()
        if user_input in valid_options:
            return user_input
        print(
            f"Invalid option. Please choose from: {', '.join(valid_options)}"
        )


def generate_advice(season: str, plant_type: str) -> str:
    """
    Generate gardening advice based on the season and plant type.

    Args:
        season (str): Current season (e.g., 'summer', 'winter').
        plant_type (str): Type of plant (e.g., 'flower', 'vegetable').

    Returns:
        str: Gardening advice for the specified season and plant type.
    """
    advice_dict = {
        "summer": {
            "flower": (
                "Water your plants regularly, "
                "provide shade, and use fertilizer to encourage blooms."
            ),
            "vegetable": (
                "Water your plants regularly keep an eye out for pests!"
            ),
        },
        "winter": {
            "flower": (
                "Protect your plants from frost with covers,"
                "Also use fertilizer sparingly."
            ),
            "vegetable": (
                "Protect your plants from frost,"
                "Monitor for pests indoors."
            )
        },
    }

    return advice_dict.get(season, {}).get(
        plant_type, "No specific advice for this combination."
    )


def main() -> None:
    """
    Main function to run the Gardening Advice App.
    Prompts the user for season and plant type and displays advice.
    """
    print("🌱 Welcome to the Gardening Advice App! 🌱\n")

    seasons = ["summer", "winter"]
    plant_types = ["flower", "vegetable"]

    # Get user input
    season = get_user_input(
        "Enter the current season (summer/winter): ", seasons
    )
    plant_type = get_user_input(
        "Enter the type of plant (flower/vegetable): ", plant_types
    )

    # Generate and display advice
    advice = generate_advice(season, plant_type)
    print("\nYour Gardening Advice:")
    print(advice)


if __name__ == "__main__":
    main()
